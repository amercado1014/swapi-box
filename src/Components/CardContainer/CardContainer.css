.card-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
}

.select {
  color: #feda4a;
  font-size: 45px;
}