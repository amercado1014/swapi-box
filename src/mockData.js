const mockCardData = {
  name: "Luke Skywalker",
  class: "people-card",
  data: {
    homeworld: "Tatooine",
    language: "Galactic Basic",
    population: "200000",
    species: "Human"
  }
};

const mockCardContainerData = [
  {
    name: "Luke Skywalker",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "Darth Vader",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "C-3PO",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "n/a",
      population: "200000",
      species: "Droid"
    }
  }
];

export {
  mockCardData,
  mockCardContainerData
};