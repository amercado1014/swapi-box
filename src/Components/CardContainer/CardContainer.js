import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({cards, setFavorites}) => {
  const displayCards = cards.map(card => {
    return <Card 
      key={card.name} 
      info={card} 
      setFavorites={setFavorites} />;
  });
  return (
    <div>
      {displayCards}
    </div>
  );
};

CardContainer.propTypes = {
  cards: PropTypes.array,
  setFavorites: PropTypes.func
};

export default CardContainer;