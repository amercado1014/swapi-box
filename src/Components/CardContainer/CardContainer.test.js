import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';
import { mockCardContainerData } from '../../mockData';

describe('Card', () => {
  let wrapper;
  let mockSetFavorites;

  beforeEach(() => {
    mockSetFavorites = jest.fn();
    wrapper = shallow(<CardContainer cards={mockCardContainerData}
      setFavorites={mockSetFavorites} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});