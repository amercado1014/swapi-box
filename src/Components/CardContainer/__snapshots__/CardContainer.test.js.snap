// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
        Array [
            Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "population": "200000",
                "species": "Human",
              },
              "name": "Luke Skywalker",
            },
            Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "population": "200000",
                "species": "Human",
              },
              "name": "Darth Vader",
            },
            Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "n/a",
                "population": "200000",
                "species": "Droid",
              },
              "name": "C-3PO",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "class": "people-card",
                              "data": Object {
                                "homeworld": "Tatooine",
                                "language": "Galactic Basic",
                                "population": "200000",
                                "species": "Human",
                              },
                              "name": "Luke Skywalker",
                            }
          }
          setFavorites={[Function]}
/>,
        <Card
          card={
                    Object {
                              "class": "people-card",
                              "data": Object {
                                "homeworld": "Tatooine",
                                "language": "Galactic Basic",
                                "population": "200000",
                                "species": "Human",
                              },
                              "name": "Darth Vader",
                            }
          }
          setFavorites={[Function]}
/>,
        <Card
          card={
                    Object {
                              "class": "people-card",
                              "data": Object {
                                "homeworld": "Tatooine",
                                "language": "n/a",
                                "population": "200000",
                                "species": "Droid",
                              },
                              "name": "C-3PO",
                            }
          }
          setFavorites={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "class": "people-card",
            "data": Object {
              "homeworld": "Tatooine",
              "language": "Galactic Basic",
              "population": "200000",
              "species": "Human",
            },
            "name": "Luke Skywalker",
          },
          "setFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Darth Vader",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "class": "people-card",
            "data": Object {
              "homeworld": "Tatooine",
              "language": "Galactic Basic",
              "population": "200000",
              "species": "Human",
            },
            "name": "Darth Vader",
          },
          "setFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "C-3PO",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "class": "people-card",
            "data": Object {
              "homeworld": "Tatooine",
              "language": "n/a",
              "population": "200000",
              "species": "Droid",
            },
            "name": "C-3PO",
          },
          "setFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "class": "people-card",
                                    "data": Object {
                                      "homeworld": "Tatooine",
                                      "language": "Galactic Basic",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    "name": "Luke Skywalker",
                                  }
            }
            setFavorites={[Function]}
/>,
          <Card
            card={
                        Object {
                                    "class": "people-card",
                                    "data": Object {
                                      "homeworld": "Tatooine",
                                      "language": "Galactic Basic",
                                      "population": "200000",
                                      "species": "Human",
                                    },
                                    "name": "Darth Vader",
                                  }
            }
            setFavorites={[Function]}
/>,
          <Card
            card={
                        Object {
                                    "class": "people-card",
                                    "data": Object {
                                      "homeworld": "Tatooine",
                                      "language": "n/a",
                                      "population": "200000",
                                      "species": "Droid",
                                    },
                                    "name": "C-3PO",
                                  }
            }
            setFavorites={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "population": "200000",
                "species": "Human",
              },
              "name": "Luke Skywalker",
            },
            "setFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Darth Vader",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "population": "200000",
                "species": "Human",
              },
              "name": "Darth Vader",
            },
            "setFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "C-3PO",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "n/a",
                "population": "200000",
                "species": "Droid",
              },
              "name": "C-3PO",
            },
            "setFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
